src/log/Entry.h

```diff
@@ -40,6 +45,7 @@ class Entry {
   time m_stamp;
   pthread_t m_thread;
   short m_prio, m_subsys;
+  std::string m_thread_name;
```

Comment:
`std::string` might be an overkill for storing 16 bytes.

Reply:
Is the `m_thread` used for anything else than obtaining thread name? If not, another alternative with lesser footprint on hot paths is available.

You could have a registry that would be mapping _unique thread ID_ -> thread name.

When a thread is spawned (I assume this would a rarely happening thing), then construct its _unique thread ID_ by incrementing some global counter and recording its name in the registry (under a lock); the obtained _unique ID_ would be then stored TLS.
An `Entry` would be stamped with the thread's _unique ID_ fetched from TLS in lock-free manner.
`dump_recent()` would lock the registry but nobody cares as it's a really cold path.

Reply:
nice idea ... thanks

Reply:
I like the lock-free part in the hot path, but if the cost difference b/w pthrad_getname_np and lock-free mechanism is minimal, would it make sense to rely on the pthread call itself?

Reply:
Yes, the question is what `pthread_getname_np()` really does inside. I can imagine an implementation that caches the names / optimizes the the _self_ case with TLS already. However, the man page says about `procfs` inspection.